### PR TITLE
swagger.yaml changes

### PR DESCRIPTION
**Pull request by pujavs**
oxAuth-Server swagger changes

### CODE REVIEW

File: docs/oxAuth_swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,2030 @@
+# raw swagger spec link:
+# https://raw.githubusercontent.com/GluuFederation/oxAuth/version_4.0/oxAuth-server/src/main/resources/swagger.yaml
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: https://raw.githubusercontent.com/GluuFederation/oxAuth/master/docs/oxAuth_swagger.yaml

File: docs/oxAuth_swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,2030 @@
...
+# https://gluu.org/swagger-ui/?url=https://raw.githubusercontent.com/GluuFederation/oxAuth-server/version_4.0/oxAuth-server/src/main/resources/swagger.yaml#/developers/setup-client
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: correct this link according to above correction

File: docs/oxAuth_swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,2030 @@
...
+openapi: 3.0.1
+info:
+  title: oxAuth-server
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: just `oxauth`

File: docs/oxAuth_swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,2030 @@
...
+  description: oxAuth-server - Authorization services
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: take description from https://github.com/GluuFederation/oxAuth/blob/master/README.md

File: docs/oxAuth_swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,2030 @@
...
+          description: Requested Claims.
+          schema:
+            type : string
+        - name: auth_req_id
+          in: query
+          required: false
+          description: CIBA authentication request Id.
+          schema:
+            type : string
+      responses:
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: the main response is 302, where is it ?

File: docs/oxAuth_swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,2030 @@
...
+        200:
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: In what case can we get 200 from /authorize endpoint ?

File: docs/oxAuth_swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,2030 @@
...
+                custom_response_headers:
+                  type : string
+                  description: Custom Response Headers.
+                claims:
+                  type : string
+                  description: Requested Claims.
+      responses:
+        200:
+          description: OK
+          content: {}
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: same as above and not 302

File: docs/oxAuth_swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,2030 @@
...
+      - username
+      - token_type
+      - exp
+      - iat
+      - sub
+      - aud
+      - iss
+      - jti
+      - acr_values
+      typ
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: Please remove it, it has nothing to do with actual REST.

Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: remove redundant new lines

Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: All this is wrong, REST API never returns anything like `oxAuth*`.

File: docs/oxAuth_swagger.yaml (hunk 1 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified the link

File: docs/oxAuth_swagger.yaml (hunk 2 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified this link

File: docs/oxAuth_swagger.yaml (hunk 3 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done

File: docs/oxAuth_swagger.yaml (hunk 4 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done

File: docs/oxAuth_swagger.yaml (hunk 5 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: I went through the code in AuthorizeRestWebServiceImpl.java and did not find 302 hence did not mention it.
302 is specified in @ApiResponses in interface AuthorizeRestWebService and thought its not in sync with the actual implementation.
Please let me know if I need correction

File: docs/oxAuth_swagger.yaml (hunk 6 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Authorize_OK_Response

File: docs/oxAuth_swagger.yaml (hunk 5 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Authorize_Error_Response

File: docs/oxAuth_swagger.yaml (hunk 7 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Authorize_OK_Response
Authorize_Error_Response

**Comment by pujavs** [comment]
> I didn't go over whole file ... It seems there are a lot of description of internal classes which doesn't make any sense. Swagger spec should describe REST request and response. It does not describe internal structure of the server and we never return anything like oxAuthClientSecretExpiresAt. Instead we return client_secret_expires_at. Please describe in swagger actual request and response, not internal structure. Also please pay attention to what is actually returned because right now I can see that there is description of responses taken from nowhere while actual responses are not described (e.g. 302 for Authorization Endpoint).

Puja - Working on removing internal classes and defining response attributes in-line

**Comment by pujavs** [comment]
Yuriy, as a sample service it will be of great help if you could let me know how to put attribute names returned by this service   ClientInfoRestWebService.java @Path("/clientinfo")
Attaching clientinfo code and snippet of response from my swagger yaml

clientinfo_code

clientinfo_response_spec

**Comment by yuriyz** [comment]
It puts into response `attributeName`:`attributeValue` pair. It's dynamic data which depends on scopes. Thus we can't say in advance how response looks. Define it as swagger object.
```
https://swagger.io/docs/specification/data-models/data-types/#object
```

File: docs/oxAuth_swagger.yaml (hunk 8 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: I have now included the relevant IntrospectionResponse attributes in-line to  /introspection operations
introspection_endpoint

**Comment by yuriyz** [comment]
Superseded by https://github.com/GluuFederation/oxAuth/pull/1285

**Pull request closed without merging**